cmc-rnaseq-de/MSSM-Penn-Pitt_ACC_DLPFC_FEATURECOUNTS_GRCh38_CQN_Sparse.Rmd

```diff
@@ -742,30 +742,32 @@ Clinical, technical and batch variables selected after phase I, II and III of mo
 writeLines(paste('Using following covariates in the model:',
                  paste(adjust.covars, collapse=', '),
                  'as fixed effects and Individual_ID is chosen as random effect'))
-  
-# Post adjusted design matrix
-DM1 = getDesignMatrix(COVARIATES[,adjust.covars,drop=F],Intercept = F)
-DM1$design = DM1$design[,linColumnFinder(DM1$design)$indepCols]
-  
-# Estimate voom weights
+
+# Post adjusted formula
+random_effect <- "Individual_ID"
+formula <- glue::glue("~ ", glue::glue_collapse(adjust.covars, sep = " + ")) %>% 
+  glue::glue(" + (1|{random_effect})")
+
+# Estimate voom weights with DREAM
 tmp = NEW.COUNTS
 tmp[is.na(tmp)] = 0
-VOOM.GENE_EXPRESSION = voom(tmp, design=DM1$design, plot=F)
+geneExpr = DGEList(tmp)
+geneExpr = edgeR::calcNormFactors(geneExpr)
 
-# Estimate consensus correlation coefficient
-correlation <- parallelDuplicateCorrelation(VOOM.GENE_EXPRESSION, block = COVARIATES$Individual_ID)
-  
-# Re-calculate voom weights with correlation of random effects
-VOOM.GENE_EXPRESSION = voom(tmp, 
-                            design=DM1$design, plot=F,
-                            block = COVARIATES$Individual_ID, 
-                            correlation = correlation$cor)
+VOOM.GENE_EXPRESSION = variancePartition::voomWithDreamWeights(counts = tmp, 
+                                                               formula = formula,
+                                                               data = COVARIATES )
 
 # Fit linear model using new weights and new design
-VOOM.GENE_EXPRESSION$E = CQN.GENE_EXPRESSION$E
-ADJUSTED.FIT = lmFit(VOOM.GENE_EXPRESSION)
-  
-# Residuals after normalisation
+# CQN NA counts to 0 counts for input to dream()
+tmp = CQN.GENE_EXPRESSION$E
```

Comment:
- Assign CQN expression values to voom object
- Assign 0 counts NA

Reply:
I found this confusing.  What you a really doing is setting NA counts to zero.  But it is not clear why.  Plus this is in log2 space. So zero is an arbitrary value.  Why is this be NA in the first place?

Reply:
[Earlier in the pipeline, counts are Winsorized](https://github.com/kelshmo/analysis/pull/2/files#diff-5447774c4084ad1b1e2b0c29720d8617R450-R492). NA is applied to counts deviating more than 3SD from the mean. It is arbitrary, but missing values are not allowed in `dream()`.

Reply:
Will address in another PR, see #4